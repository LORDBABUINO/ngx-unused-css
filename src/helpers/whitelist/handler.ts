import { Ignore } from '../../config';

export function handler(
  classes: string[],
  fileIgnore: Ignore | undefined,
  ignore: string[]
) {
  const allIgnorePatterns = [...ignore, ...(fileIgnore?.selectors || [])];

  return fileIgnore?.all
    ? []
    : classes.filter(
      (c) =>
        !allIgnorePatterns.some((ignorePattern) => c.includes(ignorePattern))
    );
}
