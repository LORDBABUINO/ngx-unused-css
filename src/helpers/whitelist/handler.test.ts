import { handler } from './handler';
import { Ignore } from '../../config';

describe('handler function', () => {
  test('filters classes based on ignore list', () => {
    const classes = ['btn-primary', 'text-center', 'hidden', 'active'];
    const ignore = ['hidden'];
    const result = handler(classes, undefined, ignore);
    expect(result).toMatchSnapshot();
  });

  test('ignores specific selectors from fileIgnore', () => {
    const classes = ['btn-primary', 'text-center', 'hidden', 'active'];
    const fileIgnore: Ignore = {
      file: 'example.css',
      selectors: ['btn-primary', 'active']
    };
    const ignore = ['hidden'];
    const result = handler(classes, fileIgnore, ignore);
    expect(result).toMatchSnapshot();
  });

  test('returns empty array if fileIgnore all is true', () => {
    const classes = ['btn-primary', 'text-center', 'hidden', 'active'];
    const fileIgnore: Ignore = { file: 'example.css', all: true };
    const ignore = ['hidden'];
    const result = handler(classes, fileIgnore, ignore);
    expect(result).toMatchSnapshot();
  });
});
