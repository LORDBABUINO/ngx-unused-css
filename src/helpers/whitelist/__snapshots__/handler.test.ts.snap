// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handler function filters classes based on ignore list 1`] = `
[
  "btn-primary",
  "text-center",
  "active",
]
`;

exports[`handler function ignores specific selectors from fileIgnore 1`] = `
[
  "text-center",
]
`;

exports[`handler function returns empty array if fileIgnore all is true 1`] = `[]`;
